### DISCUSSION TITLE
Modify HTML with middleware?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [vercel/next.js#43422] [Help]
Looking at the [available API for middleware ](https://nextjs.org/docs/api-reference/next/server)it does not seem like there is any access to generated HTML (server side or static), [however the Netlify runtime seems to be able to do this](https://www.netlify.com/blog/rewrite-html-transform-page-props-in-nextjs/).

Is there any way to access the HTML without the Netlify specific runtime, or on Vercel hosting specifically?

**Comment by icyJoseph** [comment]
If we sit as a CDN handling requests, there's roughly 4 points to execute "middleware"

- when the request arrives from the viewer
- when the request is passed to the server
- when the server passed back the response
- when the response is send back to the viewer 

I'd say Next.js' middleware is a bit more onto the first class, specially when running truly at the edge. 

Accessing the generated HTML, sounds like the last 2 rather. I am not really sure this is possible with the given tools. Also, for SPA transitions, intercepting the HTML is not necessarily possible.

**Reply by danthegoodman1** [reply]
Yeah that makes sense… a boy can dream though!

**Comment by ghost** [comment]
I have the same issue. I have 20k+ static generated pages, including their metadata. But I need to modify the `og:url` and `og:image` meta tags on request time to have the full url, i.e. to include the search params. 

This seems impossible today on Vercel. I guess I'm switching to Netlify...

**Reply by barnejs** [reply]
are you sure there's no solution? Im in a similar situation
